File: backend/src/main/java/com/example/backend/service/TaskListService.java
package com.example.backend.service;

import com.example.backend.model.TaskList;
import com.example.backend.repository.TaskListRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class TaskListService {
    @Autowired
    TaskListRepository taskListRepository;

    public TaskList insertOne(TaskList taskList) {
        return taskListRepository.save(taskList);
    }

    public void deleteOne(Long id) {
        taskListRepository.deleteById(id);
    }

    public TaskList updateOne(TaskList taskList) {
        return taskListRepository.save(taskList);
    }

    public List<TaskList> findAll(Long groupid) {
        return taskListRepository.findAllByGroupid(groupid);
    }

    public Optional<TaskList> findOne(Long id) {
        return taskListRepository.findById(id);
    }
}

File: frontend/src/api/TaskListService.ts
import Axios from "axios"

class TaskListService {
    static instance = Axios.create({
        baseURL: "http://localhost:8082/tasklist"
    })

    public static async insertOne(tasklist: TaskList) {
        let response = await this.instance.post("/", tasklist);
        return response.data
    }

    public static async deleteOne(id: number) {
        let response = await this.instance.delete("/", {
            params: {
                id
            }
        })

        return response.data
    }

    public static async updateOne(tasklist: TaskList) {
        let response = await this.instance.put("/", tasklist);
        return response.data
    }

    public static async findAll(groupid: number) {
        let response = await this.instance.get("/", {
            params: {
                groupid: groupid
            }
        })

        return response.data
    }

    public static async findOne(id: number) {
        let response = await this.instance.get(`/${id}`);
        return response.data
    }
}

export default TaskListService;